CenterForOpenScience/osf.io
# [ENG-2433] Include withdrawal requests in moderator digests

## Purpose
Put withdrawal notifications and submission notifications into the same moderator digest emails and eliminate the need for a separate template to produce the entry cards for withdrawals.

## Changes
* Fixed a bug in which we weren't storing digest emails for withdrawals
* Fixed a bug in which we were including `new_pending_withdrawal_request` events in the  generic user digest emails, thereby consuming them before they could be included in the moderator emails
* Explicitly passed the `new_pending_submissions` template when storing withdrawal requests instead of using the template inferred from the `new_pending_withdraw_requests` event.
* Updated the message when constructing withdraw messages to correctly note that withdrawal was requested instead of saying "submitted"
* Got rid of mocking in tests for `notify_moderator_registration_requests_withdrawal` that obscured the bug previously caused by the missing template for withdrawal emails.
* Wrote tests to confirm that submissions and withdrawals both appear in the moderator email digests and are not scooped up by the user digests.

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify that withdrawal requests show up alongside submissions in moderator emails
- Verify that withdrawal requests are clearly marked with "requested withdrawal"

What are the areas of risk?

Any concerns/considerations/questions that development raised?

## Documentation



## Side Effects



## Ticket
https://openscience.atlassian.net/browse/ENG-2433
